File: src/commands/SaveCommand.java
package commands;

import commands.core.ServerSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;
import organization.OrganizationManager;

public class SaveCommand extends ServerSideCommand {

    public SaveCommand(CallbackFunction callbackFunction, OrganizationManager organizationManager) {
        super(callbackFunction, organizationManager);
    }

    @Override
    protected void executeThrowableCommand(String[] args, CallbackFunction callback) {
        String filename = args[0];
        boolean isSuccessfullySaved = organizationManager.saveToFile(filename);
        callback.invoke(isSuccessfullySaved ? ExecutionStatus.SUCCESS : ExecutionStatus.FAILURE, null, filename);
    }
}

File: src/commands/server_side/SaveCommand.kt
package commands.server_side;

import OrganizationDatabase.OrganizationManagerInterface;
import network.client.udp.User;

public class SaveCommand extends ServerSideCommand {

    public SaveCommand(ServerCallbackFunction callback) {
        super(callback);
    }

    @Override
    protected void executeImplementation(
            User user,
            OrganizationManagerInterface organizationManager,
            Object[] args,
            ServerCallbackFunction callback
    ) throws Exception {
        assert args.length == 1;
        assert args[0] instanceof String;

        String filename = (String) args[0];
        callback.invoke(user, organizationManager, organizationManager.saveToFile(filename), null);
    }
}
